presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java

```diff
@@ -621,6 +621,8 @@ private AbstractRange(boolean lowerUnbounded, boolean lowerExclusive, boolean up
         protected DoubleRange(double lower, boolean lowerUnbounded, boolean lowerExclusive, double upper, boolean upperUnbounded, boolean upperExclusive, boolean nullAllowed)
         {
             super(lowerUnbounded, lowerExclusive, upperUnbounded, upperExclusive, nullAllowed);
+            checkArgument(lowerUnbounded || !Double.isNaN(lower));
```

Comment:
nit: perhaps, add a message: `checkArgument(lowerUnbounded || !Double.isNaN(lower), "lower bound is NaN");`